### PR TITLE
[Registration Form] Adding new fields (DOB, PRI / Military # / Privacy Notice)

### PR DESCRIPTION
**Pull request by fnormand01**
# Description

This PR is introducing the new registration form fields, added in the new version of the design. The following fields have been added:

- Date of birth (DD - MM ---Y)
- PRI or Military number
- Privacy Notice section (check mark + description)

All validation have also been added for each of these new fields.

## Type of change

- New feature (non-breaking change which adds functionality)

## Screenshot

**Initial Form:**

image

**Validation errors related to the added fields:**

image

**When all added fields are valid:**

image

**Privacy Notice Popup (TODO):**

image

# Testing

Manual steps to reproduce this functionality:

1.  Open the registration form from the home page.
2.  Fill out the form by adding voluntary errors (numbers in name fields, special character in DOB, etc.) and make sure that everything is working as expected.

# Checklist

Applicable for all code changes.

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new compiler warnings
- [ ] ~~My changes generate no new accessibility errors and/or warnings~~
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~
- [x] I have researched WCAG2.1 accessibility standards and met them in this PR (can be navigated using a keyboard)
- [x] My changes look good on IE 11+ and Chrome

### CODE REVIEW

File: frontend/src/components/authentication/RegistrationForm.jsx [hunk 1]
```diff
@@ -454,6 +709,45 @@ class RegistrationForm extends Component {
...
+                          LOCALIZE.authentication.createAccount.privacyNoticeLink
+                    }
+                    id="privacy-notice-checkbox"
+                    type="checkbox"
+                    style={styles.checkbox}
+                    onChange={this.changeCheckboxStatus}
+                  />
+                </div>
+                <div className="privacy-notice-grid-description">
+                  <p>
```
Comment (caleybrock) [inline frontend/src/components/authentication/RegistrationForm.jsx]: Do we need the <p> to wrap the content? It seems like it's just an extra HTML element?

File: frontend/src/components/authentication/RegistrationForm.jsx [hunk 2]
```diff
@@ -230,28 +334,50 @@ class RegistrationForm extends Component {
...
+            this.setState({ showCreatedAccountDialog: true, accountExistsError: false });
           }
         });
     }
     event.preventDefault();
   };
 
+  showPrivacyNoticePopup = event => {
+    this.setState({ showPrivacyNoticeDialog: true });
+    event.preventDefault();
```
Comment (caleybrock) [inline frontend/src/components/authentication/RegistrationForm.jsx]: Can you add a comment explaining why this is important?

Comment (fnormand01) [reply frontend/src/components/authentication/RegistrationForm.jsx]: Comments added.
See commit #https://github.com/code-for-canada/project-thundercat/pull/330/commits/a7433527072d031e5736025cb508bf7e71cf8154

File: frontend/src/components/authentication/RegistrationForm.jsx (hunk 1 above)
Comment (fnormand01) [reply frontend/src/components/authentication/RegistrationForm.jsx]: I assumed you were talking about the < p > tag.
I removed it in commit #https://github.com/code-for-canada/project-thundercat/pull/330/commits/0fa31c6f777cb4042571062857fd6fba7d44ac6e

**Comment by fnormand01** [comment]
I restructured the states order, added useful comments and created functions for aria-label conditions in order to avoid confusion. Hope it helps.

See commit #https://github.com/code-for-canada/project-thundercat/pull/330/commits/91c1d3723de0464c81acd3743d2c0a9c45288139, commit #https://github.com/code-for-canada/project-thundercat/pull/330/commits/804b255c091ded7137f8709fefb254a351c5e1e0 and commit #https://github.com/code-for-canada/project-thundercat/pull/330/commits/87b5788a8baca72f195ddcfa87305657a11795a9

**Pull request merged**